samples/bootloader/src/main.c

```diff
@@ -172,7 +172,6 @@ static void boot_from(u32_t *address)
 	CODE_UNREACHABLE;
 }
 
-void z_cstart(void) __attribute__((alias("main")));
```

Comment:
Don't we need some change in zephyr codebase for this to work? Is zephyr's z_cstart still not invoked as was the case prior to this patch?

Reply:
No, it should work out of the box. The real z_cstart is now compiled in.

Reply:
So we run the zephyr kernel initialization with this change? Do you have any numbers on code size before/after?

It is not the code size directly that worries me, more the lines of code included.

Reply:
Before: 10524 (CONFIG_KERNEL must be 'n')
After: 11628 (CONFIG_KERNEL must be 'y')

Reply:
With or without debug info?

The only interesting numbers is without any debug info IMO

Reply:
Without debug.

I think most of the extra bytes are fault handling. So it's added value I guess. It would be nice if the error handling could be scaled down though.

Reply:
Yup, but lets try to do that upstream to some degree. The cost of maintenance outweighs the cost of codesize.

Reply:
Interesting detail: The size goes down to 11292 when I comment out 

    k_thread_abort(thread);

in `z_fatal_error()`